velox/exec/Operator.cpp

```diff
@@ -665,13 +665,6 @@ void Operator::MemoryReclaimer::abort(
       !driver->state().isOnThread() || driver->state().suspended() ||
       driver->state().isTerminated);
   VELOX_CHECK(driver->task()->isCancelled());
-  if (driver->state().isOnThread() && driver->state().suspended()) {
-    // We can't abort an operator if it is running on a driver thread and
-    // suspended for memory arbitration. Otherwise, it might cause random crash
-    // when the driver thread throws after detects the aborted query.
-    return;
-  }
-
```

Comment:
CMIIW, it was previously used to handle the race condition. With this PR, it can now be removed as the race condition is addressed within the `MemoryPoolImpl::growCapacity` method. So this PR is actually an enhancement, not a fix, should we change the PR title?

Reply:
There is a race condition which can be handled by the removed check. But with the fix in this PR, we don't need this check anymore and it might improve performance as we can release the memory resource from an operator earlier. I mentioned this in PR description.